api/src/nv_ingest_api/internal/extract/pdf/engines/pdfium.py

```diff
@@ -483,6 +549,16 @@ def pdfium_extractor(
     pages_for_tables = []  # Accumulate tuples of (page_idx, np_image)
     futures = []  # To track asynchronous table/chart extraction tasks
 
+
+    rendered_imgs: Optional[List[Tuple[np.ndarray, Tuple[int, int]]]] = None
+    if (extract_tables or extract_charts or extract_infographics):
+        rendered_imgs = render_single_pdf_parallel(
+            pdf_path=pdf_stream,
+            size=(YOLOX_PAGE_IMAGE_PREPROC_WIDTH, YOLOX_PAGE_IMAGE_PREPROC_HEIGHT),
+            max_workers=pdfium_config.workers_per_progress_engine,
+            execution_trace_log=execution_trace_log,
+        )
```

Comment:
@edknv , somehow i dont see pdfium_pages_to_numpy after this change on yolox, so there is a empty wait time. Any clue why ?

Reply:
Resolved based on internal thread